utils/nctl/sh/scenarios-upgrades/upgrade_scenario_09.sh

```diff
@@ -0,0 +1,235 @@
+    local NODE_ID
+
+    log_step_upgrades 4 "Asserting nodes 1 thru 5 upgraded"
+
+    # Assert nodes are running same protocol version.
+    for NODE_ID in $(seq 1 5)
+    do
+        NX_PROTOCOL_VERSION=$(get_node_protocol_version "$NODE_ID")
+        if [ "$NX_PROTOCOL_VERSION" = "$PROTOCOL_VERSION_INITIAL" ]; then
+            log "ERROR :: upgrade failure :: nodes are not all running same protocol version"
+            log "... Node $NODE_ID: $NX_PROTOCOL_VERSION = $PROTOCOL_VERSION_INITIAL"
+            exit 1
+        else
+            log "Node $NODE_ID upgraded successfully: $PROTOCOL_VERSION_INITIAL -> $NX_PROTOCOL_VERSION"
+        fi
+    done
+}
+
+# Step 05: Assert nodes 1-5 didn't stall.
+function _step_05()
+{
+    local HEIGHT_1
+    local HEIGHT_2
+    local NODE_ID
+
+    log_step_upgrades 5 "Asserting nodes 1 thru 5 didn't stall"
+
+    HEIGHT_1=$(get_chain_height 2)
+    await_n_blocks '5' 'true' '2'
+    for NODE_ID in $(seq 1 5)
+    do
+        HEIGHT_2=$(get_chain_height "$NODE_ID")
+        if [ "$HEIGHT_2" != "N/A" ] && [ "$HEIGHT_2" -le "$HEIGHT_1" ]; then
+            log "ERROR :: upgrade failure :: node-$NODE_ID has stalled"
+            log " ... node-$NODE_ID : $HEIGHT_2 <= $HEIGHT_1"
+            exit 1
+        else
+            log " ... no stall detected on node-$NODE_ID: $HEIGHT_2 > $HEIGHT_1 [expected]"
+        fi
+    done
+}
+
+# Step 06: Await 1 era.
+function _step_06()
+{
+    log_step_upgrades 6 "awaiting next era"
+    await_n_eras '1' 'true' '5.0' '2'
+}
+
+# Step 07: Stage node 6 with old trusted hash.
+function _step_07()
+{
+    local STAGE_ID=${1}
+    local ACTIVATION_POINT=${2}
+    local HASH
+    local PATH_TO_NODE_CONFIG_UPGRADE
+    local N2_PROTO_VERSION
+
+    # Use trusted hash prior to upgrade point
+    HASH="$($(get_path_to_client) get-block -b 1 --node-address "$(get_node_address_rpc '2')" | jq -r '.result.block.hash')"
+
+    # Node 2 would be running the upgrade if we made it this far in the test.
+    # sed is for switching from: ie. 1.0.0 -> 1_0_0
+    N2_PROTO_VERSION="$(get_node_protocol_version 2 | sed 's/\./_/g')"
+
+    log_step_upgrades 7 "upgrading node 6 from stage ($STAGE_ID)"
+
+    log "... setting upgrade assets"
+
+    source "$NCTL/sh/assets/upgrade_from_stage_single_node.sh" stage="$STAGE_ID" verbose=false node="6" era="$ACTIVATION_POINT"
+    echo ""
+    # add hash to upgrades config
+    PATH_TO_NODE_CONFIG_UPGRADE="$(get_path_to_node_config $i)/$N2_PROTO_VERSION/config.toml"
+    _update_node_config_on_start "$PATH_TO_NODE_CONFIG_UPGRADE" "$HASH"
+
+    source "$NCTL/sh/node/start.sh" node='6' hash="$HASH"
+}
+
+# Step 08: Check for expected failure.
+function _step_08()
+{
+    local NODE_ID=${1}
+    local NODE_PATH
+
+    log_step_upgrades 8 "Checking for failed node-$NODE_ID..."
+
+    NODE_PATH="$(get_path_to_node $NODE_ID)"
+
+    if ( cat "$NODE_PATH"/logs/stdout.log | grep -q 'the trusted block has an older version' ); then
```

Comment:
We repeat the string `the trusted block has an older version` three times here. Please consider replacing it with a variable.